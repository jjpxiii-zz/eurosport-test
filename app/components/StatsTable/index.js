import React from 'react';
import PropTypes from 'prop-types';

import TableWithHeader from 'components/TableWithHeader';
import TableRow from 'components/TableRow';
import LoadingIndicator from 'components/LoadingIndicator';
import PlayerStatsItem from 'containers/PlayerStatsItem';

function StatsTable({ loading, error, data }) {
  if (loading) {
    return <TableWithHeader component={LoadingIndicator} />;
  }

  if (error !== false) {
    const ErrorComponent = () => (
      <TableRow item="Something went wrong, please try again!" />
    );
    return <TableWithHeader component={ErrorComponent} />;
  }

  if (data !== false) {
    return <TableWithHeader items={data} component={PlayerStatsItem} />;
  }

  return null;
}

StatsTable.propTypes = {
  loading: PropTypes.bool,
  error: PropTypes.any,
  data: PropTypes.any,
};

export default StatsTable;
