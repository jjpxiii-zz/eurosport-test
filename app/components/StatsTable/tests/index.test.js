import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { render } from 'react-testing-library';

import configureStore from '../../../configureStore';
import StatsTable from '../index';

describe('<ReposList />', () => {
  it('should render the loading indicator when its loading', () => {
    const { container } = render(<StatsTable loading />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an error if loading failed', () => {
    const { queryByText } = render(
      <IntlProvider locale="en">
        <StatsTable loading={false} error={{ message: 'Loading failed!' }} />
      </IntlProvider>,
    );
    expect(queryByText(/Something went wrong/)).not.toBeNull();
  });

  it('should render the playerStats if loading was successful', () => {
    const store = configureStore({ global: {} }, browserHistory);
    const playerStats = [
      {
        firstname: 'Stan',
        lastname: 'Wawrinka',
        shortname: 'S.WAW',
        sex: 'M',
        country: {
          picture:
            'https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png',
          code: 'SUI',
        },
        picture:
          'https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg',
        data: {
          rank: 21,
          points: 1784,
          weight: 81000,
          height: 183,
          age: 33,
          last: [1, 1, 1, 0, 1],
        },
      },
      {
        firstname: 'Rafael',
        lastname: 'Nadal',
        shortname: 'R.NAD',
        sex: 'M',
        country: {
          picture:
            'https://i.eurosport.com/_iss_/geo/country/flag/large/2203.png',
          code: 'ESP',
        },
        picture:
          'https://i.eurosport.com/_iss_/person/pp_clubteam/large/435121.jpg',
        data: {
          rank: 1,
          points: 1982,
          weight: 85000,
          height: 185,
          age: 33,
          last: [1, 0, 0, 0, 1],
        },
      },
    ];
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <StatsTable data={playerStats} error={false} />
        </IntlProvider>
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render anything if nothing interesting is provided', () => {
    const { container } = render(
      <StatsTable data={false} error={false} loading={false} />,
    );

    expect(container.firstChild).toBeNull();
  });
});
