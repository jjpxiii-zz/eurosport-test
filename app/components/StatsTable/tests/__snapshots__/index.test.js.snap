// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReposList /> should render the loading indicator when its loading 1`] = `
<div
  class="Wrapper-sc-6j63yg-0 dFyEjg"
>
  <div
    class="Wrapper-sc-12uw37d-0 bTWIVF"
  >
    <div
      class="Circle__CirclePrimitive-ww56dy-0 enQIMS"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 jCxmwS"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 dyFXB"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 ikwUeP"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 hwlQRz"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 jjXxyb"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 jVExkJ"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 haCVIz"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 ghhiLZ"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 bCbqdt"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 hZPQMA"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 fsGOmP"
    />
  </div>
</div>
`;

exports[`<ReposList /> should render the playerStats if loading was successful 1`] = `
<div
  class="Wrapper-sc-6j63yg-0 dFyEjg"
>
  <table
    class="Table-sc-281zst-0 hxgHrC"
  >
    <thead
      class="Thead-tjx0sx-0 dntDYx"
    >
      <tr>
        <th
          style="width: 110px;"
        />
        <th>
          Country
        </th>
        <th>
          Name
        </th>
        <th>
          Rank
        </th>
        <th>
          Points
        </th>
        <th>
          Height (cms)
        </th>
        <th>
          Weight (kgs)
        </th>
        <th>
          Age
        </th>
        <th>
          Last 5 matches
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="Wrapper-p1o7x3-0 bHJuyz"
      >
        <td>
          <img
            alt="Stan Wawrinka"
            src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg"
          />
        </td>
        <td>
          <img
            alt="SUI"
            class="CountryIcon-xrr94z-0 exvXay"
            src="https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png"
          />
        </td>
        <td>
          Stan
           
          Wawrinka
        </td>
        <td>
          21
        </td>
        <td>
          1784
        </td>
        <td>
          183
        </td>
        <td>
          <span>
            81
          </span>
        </td>
        <td>
          33
        </td>
        <td>
          4
          W 
          1
          L
        </td>
      </tr>
      <tr
        class="Wrapper-p1o7x3-0 bHJuyz"
      >
        <td>
          <img
            alt="Rafael Nadal"
            src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/435121.jpg"
          />
        </td>
        <td>
          <img
            alt="ESP"
            class="CountryIcon-xrr94z-0 exvXay"
            src="https://i.eurosport.com/_iss_/geo/country/flag/large/2203.png"
          />
        </td>
        <td>
          Rafael
           
          Nadal
        </td>
        <td>
          1
        </td>
        <td>
          1982
        </td>
        <td>
          185
        </td>
        <td>
          <span>
            85
          </span>
        </td>
        <td>
          33
        </td>
        <td>
          2
          W 
          3
          L
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
